### PR TITLE
Mdbenjam fix nil comment

### PR DESCRIPTION
**Pull request by mdbenjam**
If you add a comment with no text, a lot of functionality can break. In this PR we do three things:

1) Update the schema to disallow nil comments
2) Change the fake data associated with tags. It was in Fakes::AppealsRepository, but should be in seeds.rb so it doesn't run every time the server is hit.
3) Prevent creation of null comments on the frontend, and treat editing a comment to remove the text as deletion of a comment.

Connects: #1828 

**Test Plan**
- [ ] Run migration and ensure that nil comments are disallowed
- [ ] Try adding a comment with no text and make sure it's not added
- [ ] Try editing a comment so that it contains no text and make sure it goes through the delete flow

### CODE REVIEW

File: db/seeds.rb [hunk 1]
```diff
@@ -87,6 +87,15 @@ def create_annotations
     Generators::Annotation.create(comment: "This is an example comment", document_id: 2)
   end
 
+  def create_tags
```
Comment (mdbenjam) [inline db/seeds.rb]: @tejans24 I'm moving the seeded tags here so that they don't run on ever server request.

Comment (tejans24) [inline db/seeds.rb]: @mdbenjam cool, that makes sense!

File: lib/generators/tag.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class Generators::Tag
+  extend Generators::Base
+
+  class << self
+    def build(attrs)
+      attrs[:text] ||= "This is a tag"
```
Comment (NickHeiner) [inline lib/generators/tag.rb]: Maybe change this to "default generated tag text" to make it more obvious?

**Comment by NickHeiner** [comment]
The request fails when I submit a comment with only whitespace content: `'     '`:

image

Let's fix that in a later PR, though. Changes we make here could be invalidated by the work I'm doing to move comment handling to redux.

**Comment by NickHeiner** [comment]
The tests still pass, even when I undo your work with `git checkout master -- client/app`. It would be better if the tests failed if this PR wasn't applied for the frontend.

**Pull request merged**